fix: deprecate `Page.prototype.target`

jrandolf-google opened the pull request:
All methods on the `Target` instance obtained by `Page.prototype.target` are already on the `Page` class, so this getter is redundant.

OrKoN commented on test/src/page.spec.ts:
```diff
@@ -583,13 +583,9 @@ describe('Page', function () {
-        })!;
-      // 4. Connect to the popup and make sure it doesn't throw.
+      // 4. The target should always be the last one.
+      const popupTarget = page.browserContext().targets().at(-1)!;
```
let's assert that it is actually not the page target by accident

jrandolf committed:
fix: deprecate `Page.prototype.target`

All methods on `Page.prototype.target` are already on the `Page` class, so this getter is redundant.
